### PR TITLE
Java_Factorial_Program

### PR DESCRIPTION
**Pull request by Buddhimah**
Please Review my code this is a factorial program used Java

### CODE REVIEW

File: factorial/Factorial.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+
```
Comment (t2013anurag) [inline factorial/Factorial.java]: remove extra lines

File: factorial/Factorial.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        if(number==1){
+            return 1;
+
+        }
+        r= fac(number-1)*number;
+
+        return r;
+
+
```
Comment (t2013anurag) [inline factorial/Factorial.java]: remove extra lines.

File: factorial/Factorial.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    public static  int fac(int number){
+
+        int r;
+
+        if(number==1){
+            return 1;
+
+        }
+        r= fac(number-1)*number;
```
Comment (t2013anurag) [inline factorial/Factorial.java]: proper spacing b/w operands and operators.

File: factorial/Factorial.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+        int num =sc.nextInt();
+
+        if(num<0){
+            Error a = new Error("factorial for negative numbers is undefined");
+            System.out.println(a.getMessage());
+            Factorial.main(new String[1]);
+
+
+        }else {
```
Comment (t2013anurag) [inline factorial/Factorial.java]: spacing

File: factorial/Factorial.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+            System.out.println(fac(num));
+        }
+
+
```
Comment (t2013anurag) [inline factorial/Factorial.java]: -- extra lines.

Comment (Buddhimah) [reply factorial/Factorial.java]: @t2013anurag Extra spacing and the proper spacing between operands and operators are fixed 
Thank you

**Pull request merged**